content/en/docs/security/secrets-management.md

```diff
@@ -0,0 +1,276 @@
+---
+title: "Secrets Management"
+linkTitle: "Secrets Management"
+description: "Managing Secrets in a GitOps way using Flux."
+weight: 140
+---
+
+# Introduction
+
+Flux improves the application deployment process by continuously reconciling a
+desired state, defined at a source, against a target cluster. One of the challenges
+with this process is its dependency on secrets, that must not be stored in plain-text
+like the rest of the desired state.
+
+Secrets are sensitive information that an application needs to operate such as:
+credentials, passwords, keys, tokens and certificates. Managing secrets declaratively
+needs to be done right because of its broad security implications.
+
+We will cover the mechanisms supported by Flux, as well as the security principles,
+concerns and techniques to consider when managing secrets with Flux.
+
+
+## What's inside the toolbox?
+
+First of all, let's go through the different options supported by Flux and Kubernetes.
+
+Nowadays there are a multitude of secret management options. Some are available in-cluster,
+directly in the comfort of of your Kubernetes cluster, and others that are provided from
+out-of-cluster, for example a Cloud based KMS.
+
+### Kubernetes Secrets
+
+Kubernetes has a [built-in mechanism][kubernetes secrets] to store and manage secrets. The secrets
+are stored in etcd either in plain-text or [encrypted][etcd encryption].
+
+They are the vanilla offering, which is used during `flux bootstrap`, for example, to store your
+SSH Deploy Keys. Unless when the initial Flux source supports [contextual authorization],
+in which case no secrets are required.
+
+Storing plain-text secrets in your desired state is not recommended, so apart from the secret
+used to authenticate against your initial source, Flux users should not manage these. Instead,
+they should rely mostly on other mechanisms covered below.
+
+### Secrets Decryption Operators
+
+Sometimes referred to as Encrypted Secrets, Secrets Decryption Operators enable secrets to be stored
+in cipher-text as Kubernetes resources within a Flux source. They are deployed into the cluster by
+Flux in their original CustomResourceDefinition (CRD) form, which is later used by its Secret
+Decryption Operator to decrypt those secrets and generate a Kubernetes Secret.
+
+This is transparent to the consuming applications, making it a quite suitable approach to retrofit
+into an existing setup. An example of a Secret Decryption Operator is [Sealed Secrets]. 
+
+Storing encrypted secrets in Git repositories enables configuration versioning to leverage the
+same practices used for managing, versioning and releasing application and/or infrastructure
+when using declarative "everything as code", for example pull-requests, tags and branch
+strategies.
+
+Note that some sources may keep a history of the encrypted Secrets (e.g. `GitRepository`)
```

Comment:
maybe it's worth mentioning that a source containing a secret (even if encrypted) should always be private? raising this because of the following line in https://fluxcd.io/docs/guides/sealed-secrets/
> In order to store secrets safely in a public or private Git repository, you can use Bitnami’s [sealed-secrets controller](https://github.com/bitnami-labs/sealed-secrets)

same goes for the SOPS section as well.

Reply:
I am not sure we want to be that prescriptive, as each use case has different security requirements. For example, this could be a valid use amongst public repositories that share non critical secrets for example.

I tried to cover the high-level concern in:
- Access Management and Auditing: "having secrets stored (even if in encrypted form) in an easily accessible Flux source that has a loosely set of RBAC and no auditing in place may not meet such requirements."
- Summary: "ensure the appropriate level of security controls are in place"